### PR TITLE
HTML Search: filter script and style elements from search result summary text.

### LINKED ISSUE #12052: Search: don't show content of raw HTML `<script>` tags (and maybe others) in search preview
### Describe the bug

Some tags are filtered out before creating the search index (I don't know why this appears in two places in the code):

https://github.com/sphinx-doc/sphinx/blob/ae51974e217cc7590e1b9c68f18af3e043027c6d/sphinx/search/__init__.py#L221-L227

https://github.com/sphinx-doc/sphinx/blob/ae51974e217cc7590e1b9c68f18af3e043027c6d/sphinx/search/__init__.py#L487-L495

However, those parts still seem to be contained in the search preview.

### How to Reproduce

`index.rst`:

```rst
Hello
=====

.. raw:: html

    <script type="text/javascript">
      console.log("green");
    </script>

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis
nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
culpa qui officia deserunt mollit anim id est laborum.

And here the search word: green
```

When searching for "green", the JavaScript code is shown in the preview text:

image

Note that this only happens if the word "green" is also part of the "normal" text.

### Environment Information

```text
any
```

### Sphinx extensions

```python
none
```

### Additional context

This problem was reported to `nbsphinx`, where the raw HTML is created from output cells of Jupyter notebooks: https://github.com/spatialaudio/nbsphinx/issues/777

### PR DESCRIPTION
**Pull request by jayaddison**
### Feature or Bugfix
- Bugfix

### Purpose
- Filters-out the text content of `<script>` and `<style>` elements from the summary snippets displayed during search within Sphinx.

### Detail
- Also adds test coverage to demonstrate that HTML comments are not included in search result summaries.

### Relates
- Resolves #12052.

### CODE REVIEW

**Comment by picnixz** [comment]
@jayaddison Is this one fine?

File: sphinx/themes/basic/static/searchtools.js [hunk 1]
```diff
@@ -163,6 +163,8 @@ const Search = {
   htmlToText: (htmlString, anchor) => {
     const htmlElement = new DOMParser().parseFromString(htmlString, 'text/html');
     htmlElement.querySelectorAll(".headerlink").forEach((el) => { el.remove() });
```
Comment (picnixz) [inline sphinx/themes/basic/static/searchtools.js]: Create a list of selectors to remove and iterate over that list instead (easier to maintain in the future)

**Review by picnixz — APPROVED** [review]

**Comment by jayaddison** [comment]
> @jayaddison Is this one fine?

Yes, this one is ready :+1: Thank you for the review.

**Comment by picnixz** [comment]
Thank you!

**Comment by jayaddison** [comment]
Thanks again!

**Pull request merged**